kratos/utilities/delaunator_utilities.cpp

```diff
@@ -234,6 +234,77 @@ std::vector<std::size_t> ComputeTrianglesConnectivity(const std::vector<Point>&
     return connectivities;
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::pair<std::vector<std::size_t>, std::vector<double>> ComputeTrianglesConnectivity(
+    const std::vector<double>& rCoordinates,
+    const std::vector<std::array<double,2>>& rSegments,
+    const double AreaConstraint)
+{
+    // Creating the containers for the input and output
+    struct triangulateio in_mid, out_mid, vorout_mid;
+
+    InitializeTriangulateIO(in_mid);
+    InitializeTriangulateIO(out_mid);
+    InitializeTriangulateIO(vorout_mid);
+
+    // Initialize the boundary points coordinates list
+    in_mid.numberofpoints = rCoordinates.size()/2;
+    in_mid.pointlist = (REAL *) malloc(in_mid.numberofpoints * 2 * sizeof(REAL));
+
+    for(std::size_t i = 0; i < rCoordinates.size(); ++i) {
+        in_mid.pointlist[i] = rCoordinates[i];
+    }
+
+    // Initilize the segment list (note that default zero markers are assumed)
+    in_mid.numberofsegments = rSegments.size();
+    in_mid.segmentlist = (int*) malloc(in_mid.numberofsegments * 2 * sizeof(int));
+    in_mid.segmentmarkerlist = (int*) malloc(in_mid.numberofsegments * sizeof(int));
+    for (std::size_t i = 0; i < rSegments.size(); ++i) {
+        const auto& r_segment = rSegments[i];
+        in_mid.segmentlist[2*i] = r_segment[0];
+        in_mid.segmentlist[2*i + 1] = r_segment[1];
+    }
+
+    // Check https://www.cs.cmu.edu/~quake/triangle.switch.html for a detailed Triangle switches description
+    // "Q" quiet (no terminal output except errors)
+    // "q" quality mesh generation with no angles smaller than 20 degrees
+    // "p" triangulates a Planar Straight Line Graph
+    // "z" numbers all items starting from zero (rather than one)
+    // "a" imposes a maximum triangle area constrain
+    char* meshing_options;
+    std::string str = AreaConstraint > 0.0 ? "Qqpza" + std::to_string(AreaConstraint) : "Qqpz";
+    meshing_options = &str[0];
+    triangulate(meshing_options, &in_mid, &out_mid, &vorout_mid);
```

Comment:
```suggestion
    std::string meshing_options = AreaConstraint > 0.0 ? "Qqpza" + std::to_string(AreaConstraint) : "Qqpz";
    triangulate(&meshing_options[0], &in_mid, &out_mid, &vorout_mid);
```
does this work?

Reply:
I changed it according to your suggestion